core/services/vrf/delegate.go

```diff
@@ -516,30 +512,18 @@ func (lsn *listener) ProcessRequest(req *solidity_vrf_coordinator_interface.VRFC
 			"logData":        req.Raw.Data,
 		},
 	})
-	run, trrs, err := lsn.pipelineRunner.ExecuteRun(context.Background(), *lsn.job.PipelineSpec, vars, lsn.l)
-	if err != nil {
-		logger.Errorw("VRFListener: failed executing run", "err", err)
-	}
-	f := time.Now()
-	err = postgres.GormTransactionWithDefaultContext(lsn.db, func(tx *gorm.DB) error {
-		_, err = lsn.pipelineRunner.InsertFinishedRun(tx, pipeline.Run{
-			State:          pipeline.RunStatusCompleted,
-			PipelineSpecID: run.PipelineSpecID,
-			Errors:         run.Errors,
-			Outputs:        run.Outputs,
-			Meta:           run.Meta,
-			CreatedAt:      s,
-			FinishedAt:     null.TimeFrom(f),
-		}, trrs, true)
-		if err != nil {
-			return errors.Wrap(err, "VRFListener: failed to insert finished run")
-		}
+
+	run := pipeline.NewRun(*lsn.job.PipelineSpec, vars)
+	if _, err = lsn.pipelineRunner.Run(context.Background(), &run, lsn.l, true, func(tx *gorm.DB) error {
 		// Always mark consumed regardless of whether the proof failed or not.
-		return lsn.logBroadcaster.MarkConsumed(tx, lb)
-	})
-	if err != nil {
-		lsn.l.Errorw("VRFListener failed to save run", "err", err)
+		if err = lsn.logBroadcaster.MarkConsumed(tx, lb); err != nil {
```

Comment:
just to clarify: 
- if we set minConfs=0, the job not async, so the run will show up as complete in the UI  
- if we set minConfs>0, then the job is async, so the run will show up with everything complete except for the ethtx task until we get the required confs on the receipt (connected via pipelinetaskrunID in meta -> ethtx hash -> eth receipt), after which it gets updated to complete
- In both cases, we mark the log consumed as soon as the ethtx is queued

Reply:
Yeah I'm not sure how the rendering is like, but there's now a state attribute on pipeline_runs that will be set to `suspended` or `complete/errored`. Task runs that were complete up to the suspend point (we always execute everything until no more progress can be made) have their results persisted, and pending task runs are absent. On resume we reconstruct the scheduler state with the old results (plus the new data that made us resume the run) and continue executing